src/ir/module-utils.h

```diff
@@ -414,7 +414,16 @@ collectSignatures(Module& wasm,
       Counts& counts;
 
       TypeCounter(Counts& counts) : counts(counts) {}
+
       void visitExpression(Expression* curr) {
+        // A signature type could be anywhere.
```

Comment:
Since this type counting and sorting is meant to be a binary size optimization, I still think that we should only be counting the locations in which a type index needs to be written into the binary.

Reply:
Thanks, right, this is counting unnecessary things atm. Pushing a fix now.

Reply:
This turned out to be a little tricky, but I think I have it fixed now.